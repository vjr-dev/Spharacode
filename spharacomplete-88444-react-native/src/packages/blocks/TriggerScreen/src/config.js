// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.UpdatesettingAPiContentType = "application/json";
exports.UpdatesettingAPiEndPoint = "bx_block_settings/settings/update_settings";
exports.UpdatesettingAPiMethod = "PUT";
// Customizable Area End
