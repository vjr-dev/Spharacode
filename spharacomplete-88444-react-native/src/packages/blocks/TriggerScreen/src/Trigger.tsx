// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  ActivityIndicator,
  Dimensions,
  FlatList,
  Image,
  ImageBackground,
  Modal,
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { COLORS } from "../../../framework/src/Globals";
import { back1, back2, image_back, radio_button } from "./assets";
import TriggerController, { Props } from "./TriggerController";
const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;

export default class Trigger extends TriggerController {
  constructor(props: Props) {
    super(props);
  }
  renderItem(item: any) {
    return (
      <TouchableOpacity
        testID="itemButton"
        onPress={() => this.listcall(item)}
        style={{
          height: RFValue(35),
          width: "90%",
          alignSelf: "center",
          justifyContent: "center",
          borderRadius: 5,
          backgroundColor:
            this.state.Selectedname == item.aa
              ? COLORS.orangelight
              : "transparent",
        }}
      >
        <Text
          style={{
            color:
              this.state.Selectedname == item.aa ? "#000" : COLORS.lightwhite,
            marginLeft: 10,
          }}
        >
          {item.name}
        </Text>
      </TouchableOpacity>
    );
  }
  buttonTitle() {
    if(this.state.Selectedname === "power_button"){
      return "Power Button"
    }else if(this.state.Selectedname == "volume_up_button"){
      return "Volume Up Button"
    }else if(this.state.Selectedname == "volume_down_button"){
      return "Volume Down Button"
    }else{
      return "Select button type";
    }
  }
  header() {
    if(this.state.Selectedname === "power_button"){
      return "Power Button"
    }else if(this.state.Selectedname == "volume_up_button"){
      return "Volume Up Button"
    }else if(this.state.Selectedname == "volume_down_button"){
      return "Volume Down Button"
    }
  }
  renderButton(): JSX.Element {
    if (this.state.buttonlist) {
      return (
        <TouchableOpacity
          testID="actionButton"
          onPress={() => this.setState({ buttonlist: false, show: false })}
          style={styles.button2}
        >
          <Text
            style={{
              justifyContent: "center",
              color: COLORS.ultralightwhite,
              paddingHorizontal: 20,
            }}
          >
            {this.buttonTitle()}
          </Text>
        </TouchableOpacity>
      );
    } else {
      return (
        <View
          style={{
            height: 200,
            width: windowWidth - 30,
            backgroundColor: COLORS.Viewback,
            borderRadius: 25,
            marginTop: 30,
          }}
        >
          <Text
            style={{
              height: 50,
              width: "100%",
              justifyContent: "center",
              color: COLORS.ultralightwhite,
              paddingHorizontal: 20,
              paddingTop: 17,
            }}
          >
            {this.header()}
          </Text>
          <FlatList
            testID="listItem"
            data={this.state.List}
            // style={{ backgroundColor: "pink" }}
            renderItem={({ item }: any) => this.renderItem(item)}
          />
        </View>
      );
    }
  }

  renderLabel(item: any) {
    return (
      <TouchableOpacity
        testID="labelButton"
        onPress={() => this.labaleclick(item)}
        style={styles.view2}
      >
        <Text style={styles.text4}>{item.name}</Text>
        <Image
          source={radio_button}
          style={{
            height: 20,
            width: 20,
            tintColor:
              this.state.LableID == item.aa && this.state.Tempdaa == item.type
                ? COLORS.darkorange
                : COLORS.ultralightwhite,
          }}
        />
      </TouchableOpacity>
    );
  }
  selectData(): any{
    if( this.state.SelectedID === 1){
      return  this.state.data1
    }else if(this.state.SelectedID === 2){
      return this.state.data2
    }else{
      return this.state.data3
    }
  }
  render() {
    return (
      <SafeAreaView style={styles.container1}>
        <View style={styles.child}>
          <StatusBar
            animated={true}
            backgroundColor="#454545"
            barStyle={"light-content"}
            showHideTransition={"slide"}
            // hidden={true}
          />
          <ImageBackground source={back1} style={styles.image}>
            <ImageBackground source={back2} style={styles.image2}>
              <View style={styles.view1}>
                <TouchableOpacity
                  testID="backButton"
                  style={styles.button1}
                  onPress={() => this.goback()}
                >
                  <Image source={image_back} style={styles.backbutton} />
                </TouchableOpacity>
              </View>

              <Text style={styles.text1}>Trigger using button.</Text>
              <Text style={styles.text3}>
                Automatically call for help by selecting mobile button option as
                well as triggering method.
              </Text>
              {/* <Text style={styles.text3}></Text> */}
              {this.renderButton()}

              {this.state.show ? (
                <>
                  <Text style={styles.text2}>Select Trigger type.</Text>
                  <FlatList
                    testID="labelFlatList"
                    data={this.selectData()}
                    // style={{ backgroundColor: "pink" }}
                    renderItem={({ item }: any) => this.renderLabel(item)}
                  />
                </>
              ) : null}
              {/* <Text style={styles.text2}>Select Trigger type.</Text>
                                <FlatList
                                    data={this.state.updatedData}
                                    // style={{ backgroundColor: "pink" }}
                                    renderItem={({ item }: any) => {
                                      
                                        return (
                                            <TouchableOpacity
                                                onPress={() => this.labaleclick(item)}
                                                style={styles.view2}>
                                                <Text style={styles.text4}>{item.name}</Text><Image source={radio_button} style={{ height: 20, width: 20, tintColor: this.state.LableID == item.aa && this.state.Tempdaa == item.type ? COLORS.darkorange : COLORS.ultralightwhite }} />
                                            </TouchableOpacity>
                                        )
                                    }}
                                /> */}

              <Modal visible={this.state.Loader} transparent={true}>
                <View
                  style={{
                    height: windowHeight,
                    width: windowWidth,
                    alignItems: "center",
                    justifyContent: "center",
                  }}
                >
                  <View
                    style={{
                      height: 100,
                      width: windowWidth - 20,
                      alignSelf: "center",
                      justifyContent: "center",
                      alignItems: "center",
                    }}
                  >
                    <ActivityIndicator
                      animating={true}
                      size={"large"}
                      color="#f07233"
                    />
                  </View>
                </View>
              </Modal>
            </ImageBackground>
          </ImageBackground>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: windowHeight,
    width: windowWidth,
    backgroundColor: "#ffffffff",
  },
  image: {
    height: windowHeight,
    width: windowWidth,
  },
  image2: {
    height: windowHeight,
    width: windowWidth,
    alignItems: "center",
  },
  container1: { flex: 1, backgroundColor: "#454545" },
  child: { width: windowWidth },
  images: { height: 150, width: 150, alignSelf: "center", marginBottom: 20 },
  backbutton: { height: 15, width: 15, marginLeft: 20, tintColor: "#f07233" },
  view1: { height: 60, width: windowWidth, justifyContent: "center" },
  button1: { height: "100%", width: 70, justifyContent: "center" },
  logo: { height: 100, width: 100, marginTop: 40 },
  text1: {
    fontSize: 20,
    color: COLORS.white,
    marginTop: 50,
    width: windowWidth - 20,
  },
  text3: {
    fontSize: 16,
    color: COLORS.ultralightwhite,
    marginTop: 10,
    width: "90%",
  },
  text4: { fontSize: 16, color: COLORS.ultralightwhite },
  imageback: {
    height: 60,
    width: windowWidth - 25,
    marginVertical: 100,
    alignItems: "center",
    flexDirection: "row",
    backgroundColor: "#515151",
    borderRadius: 50,
  },
  ccode: {
    height: 39,
    width: 60,
    backgroundColor: "#444444",
    marginLeft: 10,
    borderRadius: 25,
    alignItems: "center",
    justifyContent: "center",
  },
  text2: {
    fontSize: 14,
    color: COLORS.orangelight,
    marginTop: 30,
    width: windowWidth - 20,
  },
  button2: {
    height: 50,
    width: windowWidth - 30,
    backgroundColor: COLORS.Viewback,
    marginTop: 30,
    borderRadius: 30,
    // alignItems: "center",
    justifyContent: "center",
  },
  view2: {
    flexDirection: "row",
    width: windowWidth - 20,
    alignSelf: "center",
    justifyContent: "space-between",
    alignItems: "center",
    height: 30,
    marginVertical: 1,
  },
});
// Customizable Area End
