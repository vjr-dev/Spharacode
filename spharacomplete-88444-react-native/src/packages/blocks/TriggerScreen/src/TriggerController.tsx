// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { BlockComponent } from "../../../framework/src/BlockComponent";
import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";
import Dashboard from "../../DashBord/src/DashbordController";
import AsyncStorage from "@react-native-async-storage/async-storage";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  Loader: boolean;
  List: any;
  Selectedname: any;
  SelectedID: any;
  buttonlist: any;
  data1: any;
  data2: any;
  data3: any;
  updatedData: any;
  show: boolean;
  LableID: any;
  Userdata: any;
  Token: any;
  Tempdaa: any;
}

interface SS {
  id: any;
}

export default class Loginscreen extends BlockComponent<Props, S, SS> {
  SetsettingID: any = "";
  private volumeInstance: Dashboard;

  constructor(props: Props) {
    super(props);
    this.volumeInstance = new Dashboard(props);
    this.receive = this.receive.bind(this);

    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.RestAPIRequestMessage),
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      getName(MessageEnum.RestAPIRequestBodyMessage),
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      getName(MessageEnum.RestAPIRequestMethodMessage),
    ];

    this.state = {
      Loader: false,
      List: [
        { id: 1, name: "Power Button", aa: "power_button" },
        { id: 2, name: "Volume Up Button", aa: "volume_up_button" },
        { id: 3, name: "Volume Down Button", aa: "volume_down_button" },
      ],
      data1: [
        {
          id: 1,
          name: "Press the power button 3 times",
          aa: "press_button_for_3_times",
          type: "power_button",
        },
        {
          id: 2,
          name: "Press the power button 5 times",
          aa: "press_button_for_5_times",
          type: "power_button",
        },
        {
          id: 3,
          name: "Press the power button 3 seconds",
          aa: "Hold_the_button_for_3_seconds",
          type: "power_button",
        },
        {
          id: 4,
          name: "Press the power button 5 seconds",
          aa: "Hold_the_button_for_5_seconds",
          type: "power_button",
        },
        { id: 5, name: "Disable", aa: "disable", type: "power_button" },
      ],
      data2: [
        {
          id: 1,
          name: "Press the volume up button 3 times",
          aa: "press_button_for_3_times",
          type: "volume_up_button",
        },
        {
          id: 2,
          name: "Press the volume up button 5 times",
          aa: "press_button_for_5_times",
          type: "volume_up_button",
        },
        {
          id: 3,
          name: "Press the volume up button 3 seconds",
          aa: "Hold_the_button_for_3_seconds",
          type: "volume_up_button",
        },
        {
          id: 4,
          name: "Press the volume up button 5 seconds",
          aa: "Hold_the_button_for_5_seconds",
          type: "volume_up_button",
        },
        { id: 5, name: "Disable", aa: "disable", type: "volume_up_button" },
      ],
      updatedData: [
        {
          id: 1,
          name: "Press the volume up/down button 3 times",
          aa: "press_button_for_3_times",
          type: "volume_up_button",
        },
        {
          id: 2,
          name: "Press the volume up/down button 5 times",
          aa: "press_button_for_5_times",
          type: "volume_up_button",
        },
        { id: 3, name: "Disable", aa: "disable", type: "volume_up_button" },
      ],
      data3: [
        {
          id: 1,
          name: "Press the volume down button 3 times",
          aa: "press_button_for_3_times",
          type: "volume_down_button",
        },
        {
          id: 2,
          name: "Press the volume down button 5 times",
          aa: "press_button_for_5_times",
          type: "volume_down_button",
        },
        {
          id: 3,
          name: "Press the volume down button 3 seconds",
          aa: "Hold_the_button_for_3_seconds",
          type: "volume_down_button",
        },
        {
          id: 4,
          name: "Press the volume down button 5 seconds",
          aa: "Hold_the_button_for_5_seconds",
          type: "volume_down_button",
        },
        { id: 5, name: "Disable", aa: "disable", type: "volume_down_button" },
      ],
      Selectedname: "Select button type",
      Userdata: "",
      Token: "",
      SelectedID: 0,
      buttonlist: true,
      show: false,
      LableID: "",
      Tempdaa: "",
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  async componentDidMount() {
    const tempToken = await AsyncStorage.getItem("Token");
    this.setState({ Userdata: this.props.route.params.data });
    this.setState({ Token: tempToken });
    this.setState({
      Selectedname: this.state.Userdata.trigger_button,
      LableID: this.state.Userdata.trigger_type,
      Tempdaa: this.state.Userdata.trigger_button,
    });
  }
  async receive(from: string, message: Message) {
    runEngine.debugLog("Message Recived", message);

    if (message.id === getName(MessageEnum.RestAPIResponceMessage)) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      let responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      if (apiRequestCallId === this.SetsettingID) {
        if (responseJson != null) {
          if (responseJson.data != null) {
            this.setState({ Loader: false });
            let response = responseJson.data.attributes;
            AsyncStorage.removeItem("VolumeButtonPressedID");
            this.updateState(response.trigger_type);
          } else {
            this.setState({ Loader: false });
            alert("ERROR");
          }
        }
      }
    }
  }
  async updateState(id: string) {
    this.volumeInstance.VolumeEvents(id);
  }
  async setclick() {
    this.setState({ Loader: true });
    const header = {
      "Content-Type": configJSON.UpdatesettingAPiContentType,
      token: this.state.Token,
    };
    const attrs = {
      trigger_button: "volume_up_button",
      trigger_type: this.state.LableID,
    };
    const data1 = {
      attributes: attrs,
    };
    const httpBody = {
      data: data1,
    };
    console.log(httpBody, "httpBody");

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.SetsettingID = requestMessage.messageId;

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.UpdatesettingAPiEndPoint
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(httpBody)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.UpdatesettingAPiMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);

    return true;
  }

  listcall(item: any) {
    this.setState({
      Selectedname: item.aa,
      SelectedID: item.id,
      buttonlist: true,
      show: true,
    });
  }
  async labaleclick(item: any) {
    this.setState({
      Selectedname: item.type,
      Tempdaa: item.type,
      LableID: item.aa,
    });
    this.setclick();
  }

  goback() {
    const { navigation }: any = this.props;
    navigation.goBack();
  }
}
// Customizable Area End
