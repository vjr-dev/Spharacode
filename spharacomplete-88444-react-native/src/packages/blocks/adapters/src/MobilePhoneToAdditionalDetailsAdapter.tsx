import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
