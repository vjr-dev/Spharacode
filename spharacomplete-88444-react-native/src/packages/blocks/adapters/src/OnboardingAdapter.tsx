import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
