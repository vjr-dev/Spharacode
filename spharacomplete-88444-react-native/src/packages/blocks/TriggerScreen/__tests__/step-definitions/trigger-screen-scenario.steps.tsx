//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";

import MessageEnum, {
  getName,
} from "../../../../framework/src/Messages/MessageEnum";
import React from "react";
import Trigger from "../../src/Trigger";
import { View } from "react-native";

const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  id: "TriggerScreen",
};

const feature = loadFeature(
  "./__tests__/features/trigger-screen-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("../../../../mobile/sample.mp3", () => ({}));
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to Trigger Screen", ({ given, when, then }) => {
    let triggerWrapper: ShallowWrapper;
    let instance: Trigger;

    given("I am a User attempting to Trigger Screen", () => {
      triggerWrapper = shallow(<Trigger {...screenProps} />);
      expect(triggerWrapper).toBeTruthy();
      expect(triggerWrapper).toMatchSnapshot();
    });

    when("I navigate to the Trigger Screen", () => {
      instance = triggerWrapper.instance() as Trigger;
    });

    then("I can press the back button", async () => {
      let backButton = triggerWrapper.findWhere(
        (node) => node.prop("testID") === "backButton"
      );
      backButton.simulate("press");
      expect(backButton).toBeTruthy();
    });

    then("I can press the power button", async () => {
      instance.setState({ buttonlist: true, Selectedname: "power_button" });
      let actionButton = triggerWrapper.findWhere(
        (node) => node.prop("testID") === "actionButton"
      );
      actionButton.simulate("press");
      expect(actionButton).toBeTruthy();
    });

    then("I can press the volume up button button", async () => {
      instance.setState({ buttonlist: true, Selectedname: "volume_up_button" });
      let actionButton = triggerWrapper.findWhere(
        (node) => node.prop("testID") === "actionButton"
      );
      actionButton.simulate("press");
      expect(actionButton).toBeTruthy();
    });

    then("I can press the volume down button button", async () => {
      instance.setState({
        buttonlist: true,
        Selectedname: "volume_down_button",
      });
      let actionButton = triggerWrapper.findWhere(
        (node) => node.prop("testID") === "actionButton"
      );
      actionButton.simulate("press");
      expect(actionButton).toBeTruthy();
    });

    then("Display the button option and i can select item", async () => {
      instance.setState({ buttonlist: false, Selectedname: "power_button" });
      let listItem = triggerWrapper.findWhere(
        (node) => node.prop("testID") === "listItem"
      );
      expect(listItem).toBeTruthy();

      let item = listItem.renderProp("renderItem")({
        item: { aa: "power_button", name: "aa", id: 1 },
      });
      expect(item).toMatchSnapshot();
      let renderItemWrapper = shallow(
        <View>
          {instance.renderItem({ aa: "power_button", name: "aa", id: 1 })}
        </View>
      );
      let itemButton = renderItemWrapper.findWhere(
        (node) => node.prop("testID") === "itemButton"
      );
      itemButton.simulate("press");
      expect(itemButton).toBeTruthy();

      item = listItem.renderProp("renderItem")({
        item: { aa: "volume_up_button", name: "aa", id: 1 },
      });
      expect(item).toMatchSnapshot();
      renderItemWrapper = shallow(
        <View>
          {instance.renderItem({ aa: "volume_up_button", name: "aa", id: 1 })}
        </View>
      );
      itemButton = renderItemWrapper.findWhere(
        (node) => node.prop("testID") === "itemButton"
      );
      itemButton.simulate("press");
      expect(itemButton).toBeTruthy();

      let labelFlatList = triggerWrapper.findWhere(
        (node) => node.prop("testID") === "labelFlatList"
      );
      expect(labelFlatList).toBeTruthy();

      instance.setState({ Tempdaa: "power_button", LableID: "power_button" });
      let item2 = labelFlatList.renderProp("renderItem")({
        item: { aa: "power_button", name: "aa", id: 1, type: "power_button" },
      });
      expect(item2).toMatchSnapshot();
      let renderLabelWrapper = shallow(
        <View>
          {instance.renderLabel({
            aa: "power_button",
            name: "aa",
            id: 1,
            type: "power_button",
          })}
        </View>
      );
      let labelButton = renderLabelWrapper.findWhere(
        (node) => node.prop("testID") === "labelButton"
      );
      labelButton.simulate("press");
      expect(labelButton).toBeTruthy();

      item2 = labelFlatList.renderProp("renderItem")({
        item: {
          aa: "volume_up_button",
          name: "aa",
          id: 1,
          type: "volume_up_button",
        },
      });
      expect(item2).toMatchSnapshot();
      renderLabelWrapper = shallow(
        <View>
          {instance.renderLabel({
            aa: "volume_up_button",
            name: "aa",
            id: 1,
            type: "volume_up_button",
          })}
        </View>
      );
      labelButton = renderLabelWrapper.findWhere(
        (node) => node.prop("testID") === "labelButton"
      );
      labelButton.simulate("press");
      expect(labelButton).toBeTruthy();
    });

    then("Update setting api work with out error", () => {
      const httpBody = {
        data: {
          attributes: {
            trigger_button: "volume_up_button",
            trigger_type: "volume_up_button",
          },
        },
      };
      const updateSettingAPI: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      updateSettingAPI.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        updateSettingAPI.messageId
      );
      updateSettingAPI.addData(
        getName(MessageEnum.RestAPIRequestBodyMessage),
        JSON.stringify(httpBody)
      );
      updateSettingAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        JSON.parse(
          JSON.stringify({
            data: { attributes: { trigger_type: "volume_up_button" } },
          })
        )
      );
      instance.SetsettingID = updateSettingAPI.messageId;
      runEngine.sendMessage("Unit Test", updateSettingAPI);
    });

    then("Update setting api work with error", () => {
      const httpBody = {
        data: {
          attributes: {
            trigger_button: "volume_up_button",
            trigger_type: "volume_up_button",
          },
        },
      };
      const updateSettingAPI: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      updateSettingAPI.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        updateSettingAPI.messageId
      );
      updateSettingAPI.addData(
        getName(MessageEnum.RestAPIRequestBodyMessage),
        JSON.stringify(httpBody)
      );
      updateSettingAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        JSON.parse(JSON.stringify({ errors: "error" }))
      );
      instance.SetsettingID = updateSettingAPI.messageId;
      runEngine.sendMessage("Unit Test", updateSettingAPI);
    });
  });
});
