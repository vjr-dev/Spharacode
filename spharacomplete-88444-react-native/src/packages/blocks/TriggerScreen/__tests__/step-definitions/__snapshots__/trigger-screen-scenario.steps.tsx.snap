// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trigger Screen User navigates to Trigger Screen 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/TriggerScreen/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/TriggerScreen/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "height": "100%",
                "justifyContent": "center",
                "width": 70,
              }
            }
            testID="backButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/TriggerScreen/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "marginLeft": 20,
                  "tintColor": "#f07233",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 20,
              "marginTop": 50,
              "width": 730,
            }
          }
        >
          Trigger using button.
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 16,
              "marginTop": 10,
              "width": "90%",
            }
          }
        >
          Automatically call for help by selecting mobile button option as well as triggering method.
        </Text>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "#515151",
              "borderRadius": 30,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 30,
              "width": 720,
            }
          }
          testID="actionButton"
        >
          <Text
            style={
              Object {
                "color": "#a1a1a1",
                "justifyContent": "center",
                "paddingHorizontal": 20,
              }
            }
          >
            Select button type
          </Text>
        </TouchableOpacity>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;

exports[`Trigger Screen User navigates to Trigger Screen 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "rgba(255,211,154,1)",
      "borderRadius": 5,
      "height": 69,
      "justifyContent": "center",
      "width": "90%",
    }
  }
  testID="itemButton"
>
  <Text
    style={
      Object {
        "color": "#000",
        "marginLeft": 10,
      }
    }
  >
    aa
  </Text>
</TouchableOpacity>
`;

exports[`Trigger Screen User navigates to Trigger Screen 3`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "transparent",
      "borderRadius": 5,
      "height": 69,
      "justifyContent": "center",
      "width": "90%",
    }
  }
  testID="itemButton"
>
  <Text
    style={
      Object {
        "color": "#cdcdcd",
        "marginLeft": 10,
      }
    }
  >
    aa
  </Text>
</TouchableOpacity>
`;

exports[`Trigger Screen User navigates to Trigger Screen 4`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "space-between",
      "marginVertical": 1,
      "width": 730,
    }
  }
  testID="labelButton"
>
  <Text
    style={
      Object {
        "color": "#a1a1a1",
        "fontSize": 16,
      }
    }
  >
    aa
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/TriggerScreen/assets/radio-button.png",
      }
    }
    style={
      Object {
        "height": 20,
        "tintColor": "rgba(255,138,73,1)",
        "width": 20,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Trigger Screen User navigates to Trigger Screen 5`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "space-between",
      "marginVertical": 1,
      "width": 730,
    }
  }
  testID="labelButton"
>
  <Text
    style={
      Object {
        "color": "#a1a1a1",
        "fontSize": 16,
      }
    }
  >
    aa
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/TriggerScreen/assets/radio-button.png",
      }
    }
    style={
      Object {
        "height": 20,
        "tintColor": "#a1a1a1",
        "width": 20,
      }
    }
  />
</TouchableOpacity>
`;
